### PR TITLE
Upgrade react scripts+react to latest versions

### PR DESCRIPTION
**Pull request by cmdcolin**
In order to avoid letting our dependencies get stale on us, I tried upgrading react-scripts and react...

This I think had been attempted before, but I tried again just to see...seemed to work

1) yarn upgrade-interactive react-scripts --latest -> up to 4.0.3
1) skip preflight check in .env file of jbrowse-web as this introduces alternative version of jest that conflicts with tsdx's built in version
1) update worker loader to not manually refer to the oneOf[1].plugins since this crashed startup, this also seemed unneeded so line was removed
1) fixed a couple lint errors that prevented startup, including adding eslint-config-react-app to root, and removing @typescript-eslint/camelcase stuff which may be due to similar stuff to here (https://github.com/typescript-eslint/typescript-eslint/issues/2077)
1) manually upgrade react and react-dom in package.json of root, jbrowse-web, jbrowse-desktop as yarn upgrade-interactive complained about @babel/preset-env

### CODE REVIEW

**Comment by cmdcolin** [comment]
close but still fails some odd lint rules on CI that dont fail locally

"Definition for rule '@typescript-eslint/no-redeclare' was not found  @typescript-eslint/no-redeclare"

no mention of this in our codebase...random suspect is eslint-config-airbnb-typescript so may need to upgrade or remove it

**Comment by cmdcolin** [comment]
this appears to generate a storybook build memory failure on CI

works locally on my machine, but on CI it produces this

Screenshot from 2021-04-29 12-46-01

**Comment by cmdcolin** [comment]
The normal full repo build also appears to fail...killed after two hours running

It would be good to get the upgrades but needs some work to make sure CI passes

**Comment by cmdcolin** [comment]
This PR actually now passes CI. Previously it was timing out CI but only when run build under node 10, but running under node 14 works, so I updated to that. I don't think we have a strict need to make our CI run under node 10, so this should be ok.


It is a bit of a crazy thing to make upgrades like this, but I think they can be important to avoid getting stale in the long run

**Comment by cmdcolin** [comment]
Note that, if merged, you may have to rm -rf node_modules and re-run yarn. If not, you may (i got this at least) get this jest error when running tests

```
❯ yarn test
yarn run v1.22.10
$ DEBUG_PRINT_LIMIT=0 jest
TypeError: Cannot read property 'onlyChanged' of undefined
    at _default (/home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/getChangedFilesPromise.js:49:20)
    at /home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/cli/index.js:296:71
    at Generator.next (<anonymous>)
    at asyncGeneratorStep (/home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/cli/index.js:108:24)
    at _next (/home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/cli/index.js:128:9)
    at /home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/cli/index.js:133:7
    at new Promise (<anonymous>)
    at /home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/cli/index.js:125:12
    at _run (/home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/cli/index.js:374:20)
    at /home/cdiesh/src/jbrowse-components/node_modules/jest-cli/node_modules/@jest/core/build/cli/index.js:178:13

```

**Comment by cmdcolin** [comment]
Note that this is a large diff because I remove airbnb-typescript

It contains many rules that in a large number of cases, we just ignored. Therefore, this removes a lot of ignores, and it also  fixes some usage of curlies that the previous PR (#1925 ) didn't catch maybe since this is upgraded to a newer eslint version

**Comment by cmdcolin** [comment]
One benefit from this would be faciliating upgrading @material-ui/data-grid, which may be useful due to cell editing functionality https://github.com/mui-org/material-ui-x/blob/master/CHANGELOG.md#400-alpha26 (it is otherwise difficult to upgrade @material-ui/data-grid currently due to #1848 )

See https://github.com/GMOD/jbrowse-components/tree/upgrade_react_scripts_datagrid which upgrades @material-ui/data-grid based off this branch successfully

**Comment by cmdcolin** [comment]
requested review to see if this is of interest, especially since it could regards data-grid?

**Comment by peterkxie** [comment]
yes definitely interesting in upgrading data-grid, since cell editing is included in the newer versions of datagrid

File: .eslintrc.base.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
     "browser": true
```
Comment (garrettjstevens) [inline .eslintrc.base.json:3]: It looks like with these changes, there are several rules that are turned off that don't affect anything anymore. I was able to delete these rules and the lint check still passed:
```diff
diff --git a/.eslintrc.base.json b/.eslintrc.base.json
--- a/.eslintrc.base.json
+++ b/.eslintrc.base.json
@@ -19,33 +19,21 @@
   },
   "plugins": ["@typescript-eslint", "eslint-plugin-tsdoc"],
   "rules": {
-    "@typescript-eslint/await-thenable": "off",
     "@typescript-eslint/ban-ts-ignore": "off",
     "@typescript-eslint/ban-ts-comment": "off",
     "@typescript-eslint/ban-types": "off",
     "@typescript-eslint/camelcase": "off",
     "@typescript-eslint/explicit-function-return-type": "off",
     "@typescript-eslint/explicit-module-boundary-types": "off",
-    "@typescript-eslint/explicit-member-accessibility": "off",
     "@typescript-eslint/no-empty-function": "off",
     "@typescript-eslint/no-unused-vars": "off",
-    "@typescript-eslint/no-misused-promises": "off",
     "@typescript-eslint/no-use-before-define": "off",
     "@typescript-eslint/no-var-requires": "off",
-    "@typescript-eslint/prefer-regexp-exec": "off",
-    "@typescript-eslint/require-await": "off",
-    "@typescript-eslint/unbound-method": "off",
-    "class-methods-use-this": "off",
     "@typescript-eslint/no-redeclare": "off",
-    "global-require": "off",
     "curly": "error",
     "no-global-assign": "warn",
     "react/no-danger": "warn",
-    "import/no-cycle": "off",
     "import/no-anonymous-default-export": "off",
-    "import/prefer-default-export": "off",
-    "jsx-a11y/mouse-events-have-key-events": "off",
-    "max-classes-per-file": "off",
     "monorepo/no-internal-import": "off",
     "no-console": [
       "warn",
@@ -54,24 +42,13 @@
       }
     ],
     "no-debugger": "warn",
-    "no-param-reassign": "off",
-    "no-plusplus": "off",
-    "no-restricted-syntax": "off",
     "no-undef": "error",
     "no-underscore-dangle": "warn",
-    "no-use-before-define": "off",
-    "prettier/prettier": "off",
     "react/destructuring-assignment": "error",
-    "react/forbid-prop-types": "off",
-    "react/jsx-filename-extension": "off",
-    "react/jsx-no-bind": "off",
-    "react/jsx-props-no-spreading": "off",
-    "react/no-array-index-key": "off",
     "react/no-unused-prop-types": "error",
     "react/no-unused-state": "error",
     "react/prefer-stateless-function": "error",
     "react/prop-types": "error",
-    "react/require-default-props": "off",
     "spaced-comment": [
       "error",
       "always",
```

**Review by garrettjstevens — COMMENTED** [review]
I think keeping up with react-scripts is a good idea, and the lint changes are good, too. I left a couple inline comments, and then also:

If you change the jest version in the devDependencies of `products/jbrowse-web/package.json`, you can avoid having to set `SKIP_PREFLIGHT_CHECK` to true. I was able to use desktop fine without setting that as well, so I'd advocate for removing both `.env` files.

**Comment by cmdcolin** [comment]
interesting, I thought I had tried basically that (I had ^26 in dev dependency instead of explicitly 26.6.0) tried clearing that up and seems like it worked

File: products/jbrowse-web/src/tests/JBrowse.test.js [hunk 2]
```diff
@@ -44,6 +43,8 @@ setup()
 
 afterEach(cleanup)
 
+const delay = { timeout: 10000 }
```
[the remark this answers is not in the document]
Comment (cmdcolin) [reply products/jbrowse-web/src/tests/JBrowse.test.js]: spose so, I really only want to specify delay here but renamed anyways

**Review by garrettjstevens — APPROVED** [review]

**Comment by cmdcolin** [comment]
marking as non-draft

**Comment by cmdcolin** [comment]
@garrettjstevens I added skipLibCheck to the jbrowse-cli also now..I found that the jbrowse-cli gave errors with @types/react when I updated root package.json @types/react to 17. 

This skipLibCheck is an option that is true in our root config

Short explainer on stackoverflow https://stackoverflow.com/questions/52311779/usage-of-the-typescript-compiler-argument-skiplibcheck

It seems like it relates to declare.d.ts files...I think this should be ok for cli...the cli doesn't even use react hehe

**Comment by cmdcolin** [comment]
xref https://github.com/GMOD/jbrowse-components/pull/1930/commits/058017baf2a57cff0e1351d3652f19e68f21b17c

**Comment by cmdcolin** [comment]
this should be good to go now :)

**Pull request merged**